contents/javascript/Javascript-framework-React.md

```diff
@@ -0,0 +1,134 @@
+
+<br/>
+
+With components written in JavaScript, you can easily pass data through your application and keep state out of the DOM. You can read more about components and props in the React [documentation](https://reactjs.org/docs/components-and-props.html). 
+
+#### React is Fast
+
+Web applications can have a lot of user interaction and data updates. Extensive DOM manipulation can be a performance bottleneck in these applications as the DOM is tree-structured - simple changes at the top can trigger the entire application to get updated. 
+
+React solves this problem by using a virtual DOM. The virtual DOM is a JavaScript object that is kept in the memory of your application. In other words, it is a virtual representation of the actual DOM.
+
+Any updates to your application will first be made to the virtual DOM. Then, React will compare the virtual DOM with the actual DOM using a [diffing algorithm](https://reactjs.org/docs/reconciliation.html) to find out what changes have to be made to the actual DOM. 
+Finally, React figures out how to update the actual DOM in the most effective way possible to guarantee minimum update time. This results in better performance and a cleaner user experience, even with data-intensive applications.
+
+Those are the three main benefits of React. Other advantages are as follows:
+ - React only allows data to flow downwards (one-way data binding), which makes your application easier to debug.
+ - It is a constantly developing open-source library that is backed by Facebook, so you can be sure of getting community support.
+ - If your application is large, you can manage its state conveniently using [Redux](https://redux.js.org/introduction/getting-started). Note that Redux can be used with any other view library, but it is mostly used with React. Hence, there are more resources for learning React + Redux.
+
+### Why not React?
+
+Like any other library, React has its disadvantages, which are as follows:
+ - The high pace of development means that you would need to regularly relearn how to do things.
+ - The huge number of tools you can integrate with React don't always have comprehensive documentation.
+ - React is just a UI library. As React only allows one-way data binding, you would need to use [Flux](https://github.com/facebook/flux), a new application architecture introduced by Facebook that favours unidirectional data flow. 
+ However, if you prefer working with the MVC pattern, you should consider a framework that lets you use it.
+
+#### React and other Competing Alternatives
+
+There are a lot of JavaScript frameworks and libraries that you can use to build your next web application. Some popular alternatives to React are [Angular](https://angular.io/) and [Vue](https://vuejs.org/).
```

Comment:
Link to the corresponding book chapters instead of external websites?

Reply:
As of now, the book chapter on Angular hasn't been added yet. 
I could link to the chapter on `Vue` though.